src/plugins/dashboard/public/application/hooks/use_dashboard_app_state.ts

```diff
@@ -231,16 +231,23 @@ export const useDashboardAppState = ({
       /**
        * Build the dashboard container embeddable, and apply the incoming embeddable if it exists.
        */
+
       const dashboardContainer = await buildDashboardContainer({
         ...dashboardBuildContext,
         initialDashboardState,
         incomingEmbeddable,
         savedDashboard,
         data,
         executionContext: {
+          type: 'dashboard',
           description: savedDashboard.title,
+          meta: {
+            viewMode: initialDashboardState.viewMode,
```

Comment:
The problem with this is that it only captures the original dashboard state (when the app got loaded). Changing the view mode and refreshing will still count as the original view mode. Can we somehow make this dynamic so it relates to the current mode rather than the initial one?

Reply:
The `initialDashboardState` at this point actually contains the correct state, combined with the session app state, the url app state etc. So if you change the view mode then **reload** the page, this will show the correct view mode. 

That said, this is not dynamic in the sense that it would update any time the dashboard state changes, i.e. you _need_ to reload the page for the changes to be reflected. So if you loaded the lens panels initially in view mode, then switched to edit mode, the events would be reported as `render_lens_vis_dashboard_view` potentially leading to undercounting edit mode. How much of a problem is this?

Reply:
@ghudgins before we go into more complex implementations - is differentiating dashboard edit and view mode something that’s important or is it ok to leave it out for the first version?

Reply:
I agree, we can leave it out